x-pack/plugin/deprecation/src/internalClusterTest/java/org/elasticsearch/http/plugin/TestDeprecationHeaderRestAction.java

```diff
@@ -1,22 +1,9 @@
 /*
- * Licensed to Elasticsearch under one or more contributor
- * license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright
- * ownership. Elasticsearch licenses this file to you under
- * the Apache License, Version 2.0 (the "License"); you may
- * not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
  */
-package org.elasticsearch.http;
+package org.elasticsearch.http.plugin;
```

Comment:
This should probably be `org.elasticsearch.xpack.deprecation`?

Reply:
I moved this to `org.elasticsearch.xpack.deprecation.plugin`. I felt that keeping the `plugin` part was ever so slightly nicer since those classes are indeed loaded as an ES plugin, and don't contain the test code itself. However I can change that again if you feel strongly about it.

Reply:
I don't feel strongly, but I think we have overused `plugin` as a package suffix. The `o.e.x.deprecation` is meaningful on its own, and tests living in the same package as the plugin they are testing seems valid.

Reply:
On reflection I agree with you, so I've moved the classes. Having a proliferation of "plugin" packages doesn't seem useful.